version: '3.9'
services:
    app:
        build:
            target: ${DJANGO_ENV:-production}
            context: .
            dockerfile: Dockerfile
            args:
                - DJANGO_ENV=${DJANGO_ENV:-production}
        ports:
            - '${APP_PORT:-8000}:80'
        volumes:
            - '.:/app'
        networks:
            - net
        command: python -Wd manage.py runserver 0.0.0.0:80
        restart: always
    db:
        container_name: phonebook-postgres-db
        image: 'postgres:16'
        ports:
            - '${FORWARD_DB_PORT:-5432}:5432'
        environment:
            PGPASSWORD: '${DB_PASSWORD:-secret}'
            POSTGRES_DB: '${DB_DATABASE}'
            POSTGRES_USER: '${DB_USERNAME}'
            POSTGRES_PASSWORD: '${DB_PASSWORD:-secret}'
        volumes:
            - 'db_storage:/var/lib/postgresql/data'
        networks:
            - net
        healthcheck:
            test: ["CMD", "pg_isready", "-q", "-d", "${DB_DATABASE}", "-U", "${DB_USERNAME}"]
            retries: 3
            timeout: 5s
        restart: always
networks:
    net:
        driver: bridge
volumes:
    db_storage:
        driver: local
